### PR TITLE
Change Get-Help cmdlet -Parameter parameter so it accepts string arrays.

### LINKED ISSUE #8453: Allow Get-Help cmdlet -Parameter parameter to accept string array.
# Summary of the new feature/enhancement

It would be useful, if the **-Parameter** parameter of the `Get-Help` cmdlet accept string array,
so we can use commands like

```powershell
Get-Help -Name Get-Command -Parameter Verb, Noun
```

### PR DESCRIPTION
**Pull request by sethvs**
## PR Summary
Fix #8453 

This PR changes `Get-Help` cmdlet **-Parameter** parameter so it accepts string arrays.
Also changes `help` function.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 1]
```diff
@@ -157,7 +157,12 @@ public SwitchParameter Examples
         /// Support WildCard strings as supported by WildcardPattern class.
         /// </remarks>
         [Parameter(ParameterSetName = "Parameters", Mandatory = true)]
-        public string Parameter { set; get; }
+        public string[] Parameter
+        {
+            get => _parameters;
+            set => _parameters = value;
+        }
+        private string[] _parameters;
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommands.cs]: Why we use the private variable? Why not `public string[] Parameter { set; get; }`

**Review by iSazonov — COMMENTED** [review]
We need PowerShell Committee approval to change the public API.

File: src/System.Management.Automation/help/HelpCommands.cs (hunk 1 above)
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: Agree.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by iSazonov** [comment]
@sethvs Please fix style issues (excluding complex method).

**Comment by sethvs** [comment]
@iSazonov For some reason CodeFactor still shows lines with issues from the first commit. There are no such lines now.

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 2]
```diff
@@ -550,9 +560,20 @@ private void WriteObjectsOrShowOnlineHelp(HelpInfo helpInfo, bool showFullHelp)
                     }
                     else
                     {
-                        if (!string.IsNullOrEmpty(Parameter))
+                        if (Parameter != null)
                         {
-                            PSObject[] pInfos = helpInfo.GetParameter(Parameter);
+                            List<PSObject> pInfosList = new List<PSObject>(Parameter.Length);
+
+                            foreach (var parameter in Parameter)
```
[the remark this answers is not in the document]
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: Done.

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 3]
```diff
@@ -476,7 +497,7 @@ private void ValidateAndThrowIfError(HelpCategory cat)
 
             if ((cat & supportedCategories) == 0)
             {
-                if (!string.IsNullOrEmpty(Parameter))
+                if (Parameter != null)
```
[the remark this answers is not in the document]
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: **Parameter** is a string array now.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee reviewed this and is ok with the change

File: src/System.Management.Automation/help/HelpCommands.cs [hunk 4]
```diff
@@ -430,15 +430,36 @@ private PSObject TransformView(PSObject originalHelpObject)
...
 
         /// <summary>
-        /// Gets the parameter info for patterns identified Parameter property.
+        /// Gets the parameter info for patterns identified by Parameter property.
+        /// </summary>
+        /// <param name="helpInfo">HelpInfo Object to look for the parameter.</param>
+        /// <returns>Array of parameter infos.</returns>
+        private PSObject[] GetParameterInfo(HelpInfo helpInfo)
+        {
+            List<PSObject> parameterInfosList = new List<PSObject>(Parameter.Length);
```
[the remark this answers is not in the document]
Comment (sethvs) [reply src/System.Management.Automation/help/HelpCommands.cs]: I'm not sure that this is necessary.

**Comment by adityapatwardhan** [comment]
@sethvs Can you please add tests for the new functionality?

**Comment by sethvs** [comment]
Strange error (no failed tests) with macOS check.
Close and reopen to restart.

**Comment by sethvs** [comment]
macOS check throws **Dependency precheck failed!**
Again, no failed tests.
Guys, can you take a look into it?

**Comment by adityapatwardhan** [comment]
Having a look.

**Comment by adityapatwardhan** [comment]
It seems that Bootstrap is failing with the following issue: @TravisEz13 any idea?

```
2019-01-15T17:51:59.5248120Z Warning: openssl 1.0.2q is already installed and up-to-date
2019-01-15T17:51:59.5248480Z To reinstall 1.0.2q, run `brew reinstall openssl`
2019-01-15T17:51:59.5424690Z Error: cmake 3.13.2 is already installed
2019-01-15T17:51:59.5424850Z To upgrade to 3.13.3, run `brew upgrade cmake`
2019-01-15T17:52:01.9032320Z Warning: curl 7.63.0 is already installed and up-to-date
2019-01-15T17:52:01.9032780Z To reinstall 7.63.0, run `brew reinstall curl`
2019-01-15T17:52:06.2156000Z Successfully installed cabin-0.9.0
2019-01-15T17:52:06.2156670Z Successfully installed backports-3.11.4
2019-01-15T17:52:06.2157410Z Successfully installed arr-pm-0.0.10
2019-01-15T17:52:06.2157950Z Successfully installed clamp-1.0.1
2019-01-15T17:52:06.2158020Z Building native extensions. This could take a while...
2019-01-15T17:53:06.0164980Z Before reporting this, could you check that the file you're documenting
2019-01-15T17:53:06.0165390Z has proper syntax:
2019-01-15T17:53:06.0165830Z 
2019-01-15T17:53:06.0166810Z   /usr/local/opt/ruby/bin/ruby -c lib/childprocess/tools/generator.rb
2019-01-15T17:53:06.0167080Z 
2019-01-15T17:53:06.0167210Z RDoc is not a full Ruby parser and will fail when fed invalid ruby programs.
2019-01-15T17:53:06.0167280Z 
2019-01-15T17:53:06.0167360Z The internal error was:
2019-01-15T17:53:06.0167540Z 
2019-01-15T17:53:06.0168380Z 	(NoMethodError) undefined method `[]' for nil:NilClass
2019-01-15T17:53:06.0168460Z 
2019-01-15T17:53:06.0168880Z ERROR:  While executing gem ... (NoMethodError)
2019-01-15T17:53:06.0169910Z     undefined method `[]' for nil:NilClass
2019-01-15T17:53:06.0245480Z Successfully installed ffi-1.10.0
```

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**